src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -37,12 +46,45 @@ public LogicManager(Model model, Storage storage) {
         gradPadParser = new GradPadParser();
     }
 
+    private void assignStalledComponents(Command command, String commandText) {
+        stalledCommand = command;
+        stalledCommandText = commandText.trim();
+    }
+
     @Override
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
+        if (stalledCommand instanceof Command) {
+            if (!commandText.equalsIgnoreCase("yes")) {
+                stalledCommand = null;
+                return new CommandResult(MESSAGE_CONFIRMATION_CANCEL
+                        + String.format("\"%s\"", stalledCommandText));
+            }
+        }
+
         CommandResult commandResult;
         Command command = gradPadParser.parseCommand(commandText);
+
+        if (command instanceof ClearCommand) {
+            assignStalledComponents(command, commandText);
+            return new CommandResult(ClearCommand.MESSAGE_CONFIRMATION + MESSAGE_CONFIRMATION_SYNTAX);
+        } else if (command instanceof DeleteCommand) {
+            assignStalledComponents(command, commandText);
+            Module moduleToBeDeleted = ((DeleteCommand) command).getModuleToDelete(model);
```

Comment:
These two lines need to be swapped 😁  cos we don't want to stall the command before checking that it's valid. Right now if you type in an invalid delete e.g. "delete 100000", the command is still stalled and GradPad will wait for a confirmation. When a user then corrects himself and types "delete 1", GradPad will say "command aborted" instead of executing the corrected command